src/scenic/core/regions.py

```diff
@@ -2055,6 +2062,152 @@ def evaluateInner(self, context):
         )
 
 
+class VoxelRegion(Region):
+    """Region represented by a voxel grid in 3D space.
+
+    Args:
+        encoding: A numpy array encoding the voxel grid.
+        dimensions: An optional 3-tuple, with the values representing width, length, height
+          respectively. The voxel region will be scaled to have these dimensions. Default
+          value (1,1,1).
+        position: A position, which determines where the center of the region will be. Default
+          value is the origin, (0,0,0).
+        voxelGrid: Optionally, just pass in the final trimesh voxelGrid to be used. Passing this
+          parameter will result in encoding, dimensions, and position being ignored.
+        orientation: An optional vector field describing the preferred orientation at every point in
+          the region.
+        name: An optional name to help with debugging.
+        lazy: Whether or not to be lazy about pre-computing internal values. Set this to True if this
+          VoxelRegion is unlikely to be used outside of an intermediate step in compiling/pruning.
+    """
+
+    def __init__(self, voxelGrid, orientation=None, name=None, lazy=False):
+        # Initialize superclass
+        super().__init__(name, orientation=orientation)
+
+        # Work around Trimesh caching bug
+        self._voxelGrid = trimesh.voxel.VoxelGrid(
+            voxelGrid.encoding, transform=voxelGrid.transform.copy()
+        )
+
+        # Check that the encoding isn't empty. In that case, raise an error.
+        if self._voxelGrid.encoding.is_empty:
+            raise ValueError("Tried to create an empty VoxelRegion.")
+
+        # Transform voxel grid points and extract scale
+        self.voxel_points = self._voxelGrid.points
+        self.scale = self._voxelGrid.scale
+
+        # Initialize KD-Tree for containment checking if not lazy
+        if not lazy:
+            self.kdTree
+
+    @cached_property
+    def voxelGrid(self):
+        return self._voxelGrid
+
+    @cached_property
+    def kdTree(self):
+        return scipy.spatial.KDTree(self.voxel_points)
+
+    def containsPoint(self, point):
+        point = toVector(point)
+
+        # Find closest voxel point
+        _, index = self.kdTree.query([point])
+        closest_point = self.voxel_points[index]
+
+        # Check voxel containment
+        voxel_low = closest_point - self.scale / 2
+        voxel_high = closest_point + self.scale / 2
+
+        return numpy.all(voxel_low <= point, axis=1) & numpy.all(
+            point <= voxel_high, axis=1
+        )
```

Comment:
This returns an array containing a bool, not just a bool (better extend the test to check for that!). I think you can replace `[point]` with just `point` when calling `self.kdTree.query` and then get rid of the `axis` stuff in the last line.

Reply:
Good catch!